src/thread/thread_d.hpp

```diff
@@ -390,7 +392,8 @@ namespace fc {
               /* NB: At least on Win64, this only catches a yield while in the body of 
                * a catch block; it fails to catch a yield while unwinding the stack, which 
                * is probably just as likely to cause crashes */
-              assert(std::current_exception() == std::exception_ptr());
+              if( std::current_exception() != std::exception_ptr() )
+                 print_stacktrace( std::cerr );
```

Comment:
This will only print for Boost > 1.65. Would it be better to elog() as well?

Reply:
Done.

Reply:
How about logging the whole exception?

Reply:
Done.